### PR TITLE
Fix #9392. Allow override java defined methods which have a field with same name

### LINKED ISSUE #9392: Regression: Can't override java defined method which have variable with same name
## Minimized code

```Scala
class C extends java.util.LinkedList[String] {
  override def size(): Int = super.size()
}
```

## Output

Compile error in `0.25.0-RC2` and `0.26.0-bin-20200719-21a90aa-NIGHTLY`:
```scala
[error] -- Error: /Users/thanhbv/tmp/ex/src/main/scala/C.scala:2:15 --------------------
[error] 2 |  override def size(): Int = super.size()
[error]   |               ^
[error]   |error overriding variable size in class LinkedList of type Int;
[error]   |  method size of type (): Int cannot override final member variable size in class LinkedList
```

## Expectation
Compile successfully as in dotty `0.24.0` & in scala 2

### PR DESCRIPTION
**Pull request by ohze**
Fix #9392

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/Denotations.scala [hunk 1]
```diff
@@ -977,7 +977,7 @@ object Denotations {
           true
         case MethodNotAMethodMatch =>
           // Java allows defining both a field and a zero-parameter method with the same name
-          !ctx.erasedTypes && !(symbol.is(JavaDefined) && other.symbol.is(JavaDefined))
+          !ctx.erasedTypes && !other.symbol.is(JavaDefined)
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/core/Denotations.scala:980]: `a.matches(b)` and `b.matches(a)` should always produce the same result (it should be a symmetrical relation, like equality), so I don't think this is a good way to fix the problem.

Comment (giabao) [reply compiler/src/dotty/tools/dotc/core/Denotations.scala]: Resolved. Ready for review. @smarter

**Comment by giabao** [comment]
The change request are resolved. Pls review @smarter ❤️

**Comment by smarter** [comment]
Thanks for working on this but I still don't think this is a correct approach, I've opened a separate PR with an alternative fix: https://github.com/lampepfl/dotty/pull/9412

**Comment by smarter** [comment]
We decided to merge https://github.com/lampepfl/dotty/pull/9452 in the end

**Pull request closed without merging**